src/app/component/CarouselScroll/CarouselScroll.component.js

```diff
@@ -0,0 +1,197 @@
+        const { activeItemId, children: { length: childrenLength } } = this.props;
+        const { activeItemId: prevaAtiveItemId } = this.state;
+
+        if (prevChildrenLength !== childrenLength) {
+            this.handleReset();
+
+            return;
+        }
+
+        if (activeItemId !== null && activeItemId !== prevaAtiveItemId) {
+            this.handleChange(activeItemId);
+        }
+    }
+
+    getNextTranslate(nextId) {
+        const { showedItemCount, showedActiveItemNr, children: { length: childrenLength } } = this.props;
+        const { offsetHeight } = this.itemRef.current;
+
+        // When selected item isnt reached wanted position
+        if (nextId < (showedActiveItemNr - 1) || childrenLength <= showedItemCount) {
+            return 0;
+        }
+
+        const isEndReached = (showedItemCount - showedActiveItemNr) + nextId + 1 > childrenLength;
+
+        const possition = isEndReached
+            ? childrenLength - showedItemCount
+            : nextId - (showedActiveItemNr - 1);
+
+        return `${ -possition * offsetHeight }px`;
+    }
+
+    setTranslate(nextId) {
+        const { children: { length: childrenLength }, showedItemCount } = this.props;
+
+        if (childrenLength <= showedItemCount) {
+            return;
+        }
+
+        const translate = this.getNextTranslate(nextId);
+        CSS.setVariable(this.carouselRef, 'translateY', translate);
+    }
+
+    handleArrowClick = (isNextArrow) => {
+        const { children } = this.props;
+        const { activeItemId: prevActiveItemId } = this.state;
+
+        const activeItemId = prevActiveItemId + (isNextArrow ? 1 : -1);
+
+        if (children.length - 1 < activeItemId || activeItemId < 0) {
+            return;
+        }
+
+        this.handleChange(activeItemId);
+    };
+
+    handleChange(nextId) {
+        const { onChange } = this.props;
+
+        this.setTranslate(nextId);
+        this.setState({ activeItemId: nextId });
+        onChange(nextId);
+    }
+
+    handleReset() {
+        const { onChange } = this.props;
+
+        const activeItemId = 0;
+
+        CSS.setVariable(this.carouselRef, 'translateY', 0);
+
+        onChange(activeItemId);
+        this.setState({ activeItemId });
+    }
+
+    renderArrow(isNextArrow = false) {
+        const { showArrow, children: { length: childrenLength }, showedItemCount } = this.props;
+        const { activeItemId } = this.state;
+
+        if (!showArrow || childrenLength <= showedItemCount) {
+            return null;
+        }
+
+        const isDisabled = isNextArrow
+            ? activeItemId === childrenLength - 1
+            : !activeItemId;
+        const mods = { isNextArrow, isDisabled };
+
+        return (
+            <button
+              block="CarouselScroll"
+              elem="Arrow"
+              mods={ mods }
+              // eslint-disable-next-line react/jsx-no-bind
+              onClick={ () => this.handleArrowClick(isNextArrow) }
```

Comment:
Oh well, that is lazy. I think the better way here, would be to bind the define `this.handleNextArrowClick` and `this.handlePrevArrowClick` to `this.handleArrowClick` but using different arguments, so then you can use it here without creating and arrow function.